aws-android-sdk-appsync/src/main/java/com/amazonaws/mobileconnectors/appsync/SubscriptionAuthorizer.java

```diff
@@ -199,8 +205,34 @@ private static String getDataJson(Subscription subscription) {
         }
     }
 
-    private static String getApiUrl(AWSConfiguration awsConfiguration) throws JSONException {
-        return awsConfiguration.optJsonObject("AppSync").getString("ApiUrl");
+    private String getRegion() throws JSONException {
+         return mRegionStr != null
+                 ? mRegionStr
+                 : mAwsConfiguration
+                     .optJsonObject("CredentialsProvider")
+                     .getJSONObject("CognitoIdentity")
+                     .getJSONObject(mAwsConfiguration.getConfiguration())
+                     .getString("Region");
+    }
+
+    private String getIdentityPoolId() throws JSONException {
+         return mAwsConfiguration
+                 .optJsonObject("CredentialsProvider")
+                 .getJSONObject("CognitoIdentity")
+                 .getJSONObject(mAwsConfiguration.getConfiguration())
+                 .getString("PoolId");
+    }
+
+    private String getApiUrl() throws JSONException {
+        return mServerUrl != null
+                ? mServerUrl
+                : mAwsConfiguration.optJsonObject("AppSync").getString("ApiUrl");
+    }
+
+    private String getApiKey() throws JSONException {
+         return mApiKeyProvider != null
+                 ? mApiKeyProvider.getAPIKey()
+                 : mAwsConfiguration.optJsonObject("AppSync").getString("ApiKey");
```

Comment:
These have all been changed to non-static, and will check if their respective data has been provided before digging into the `AWSConfiguration`